pkg/kv/kvserver/replica_consistency_test.go

```diff
@@ -118,3 +124,100 @@ func TestGetChecksumNotSuccessfulExitConditions(t *testing.T) {
 	}
 	require.Nil(t, rc.Checksum)
 }
+
+// TestReplicaChecksumSHA512 checks that a given dataset produces the expected
+// checksum.
+func TestReplicaChecksumSHA512(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	// We incrementally add writes, and check the checksums after each write to
+	// make sure they differ such that each write affects the checksum.
+	testcases := []struct {
+		key    string
+		endKey string
+		ts     int32
+		value  string
+		expect string
+	}{
+		// Start with a noop write to hash the initial state.
+		{"", "", 0, "", "cf83e1357eefb8bdf1542850d66d8007d620e4050b5715dc83f4a921d36ce9ce47d0d13c5d85f2b0ff8318d2877eec2f63b931bd47417a81a538327af927da3e"},
```

Comment:
Consider making this datadriven using `echotest.Require`:

https://github.com/cockroachdb/cockroach/blob/ad59351e4b8a2581d6ce53e113b4e242ac7ebc33/pkg/testutils/echotest/echotest.go#L19-L28

That way all you have to do is to print the checksums, and let datadriven handle the rest via `-rewrite`. This is relevant since this test will break every time anything changes about the initial state, which does happen occasionally.